cmd/kubectl-gadget/deploy.go

```diff
@@ -506,12 +546,26 @@ func runDeploy(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	// modify the otel-metrics-listen-address if specified
+	if otelMetricsListenAddr != "" {
```

Comment:
Perhaps only apply this if [otelMetricsListen](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/cmd/kubectl-gadget/deploy.go#L112) is true.